### PR TITLE
OCPCLOUD-2723: Bumps k8s deps to v1.31

### PR DESCRIPTION
**Pull request by nrb**
Signed-off-by: Nolan Brubaker <nolan@nbrubaker.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,18 +13,18 @@ require (
...
+	k8s.io/client-go v0.31.1
+	k8s.io/code-generator v0.31.1
+	k8s.io/component-base v0.31.1
+	k8s.io/component-helpers v0.31.1
+	k8s.io/controller-manager v0.31.1
+	k8s.io/klog/v2 v2.130.1
+	k8s.io/kube-controller-manager v0.31.1
+	k8s.io/kubelet v0.31.1
+	k8s.io/metrics v0.31.1
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
```
Comment (JoelSpeed) [inline go.mod:28]: The upstream PR also updates this package, should we?

Comment (nrb) [reply go.mod]: Done.

**Comment by nrb** [comment]
/retest

Jobs failed on image pulls

**Review by elmiko — COMMENTED** [review]
/approve
/lgtm

**Comment by miyadav** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-gcp-master-regression-clusterinfra-cucushift-rehearse-gcp-ipi

**Pull request merged**